content/blog/super-charging-our-github-action-with-the-pulumi-automation-api/index.md

```diff
@@ -0,0 +1,200 @@
+be able to run Pulumi commands. As the Pulumi ecosystem has grown, so too did the container that GitHub Actions run.
+
+Since then, an alternative way to create a GitHub Action has become available,
+[by simply writing JavaScript](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action).
+At Pulumi, and in our community, we love expressing everything we possibly can in code, so this new support got us quite
+excited about the "better together" possibilities! [Enter Pulumi's new Automation API](https://www.pulumi.com/blog/automation-api/),
+a new feature we launched in public preview in October 2020. This new capability lets you do infrastructure as code
+&mdash; in pure code &mdash; without even needing to run a CLI. This sounded like a perfect match for JavaScript Actions.
+It allows us to programmatically invoke the right Pulumi commands in response to actions, without needing to awkwardly
+script a CLI. Instead, we were able to leverage the Automation API to fundamentally rethink our GitHub Action.
+
+### GitHub Action v2 -- Under The Hood
+
+The new version of the GitHub action is based largely on the following code:
+
+```typescript
+const main = async () => {
+  const config = await makeConfig();
+
+  if (environmentVariables.PULUMI_ACCESS_TOKEN !== '') {
+    await pulumiCli.run('login');
+  } else if (config.cloudUrl) {
+    await pulumiCli.run('login', config.cloudUrl);
+  }
+
+  const workDir = resolve(
+    environmentVariables.GITHUB_WORKSPACE,
+    config.workDir,
+  );
+
+  const stack = await LocalWorkspace.selectStack({
+    stackName: config.stackName,
+    workDir: workDir,
+  });
+
+  const onOutput = (msg: string) => {
+    .....
+  };
+
+  const actions: Record<Commands, () => Promise<string>> = {
+    up: () => stack.up({ onOutput }).then((r) => r.stdout),
+    refresh: () => stack.refresh({ onOutput }).then((r) => r.stdout),
+    destroy: () => stack.destroy({ onOutput }).then((r) => r.stdout),
+    preview: async () => {
+      const { stdout, stderr } = await stack.preview();
+      onOutput(stdout);
+      onOutput(stderr);
+      return stdout;
+    },
+  };
+  
+  const output = await actions[config.command]();
+
+  const outputs = await stack.outputs();
+
+  for (const [outKey, outExport] of Object.entries(outputs)) {
+    core.setOutput(outKey, outExport.value);
+    if (outExport.secret) {
+      core.setSecret(outExport.value);
+    }
+  }
+};
+```
+
+It doesn't have a lot of functionality as the power lays in the Automation API itself. The Action is just an enablement
+of the Automation API that can be driven from GitHub Actions. The fact that it has no real functionality, means that as
+changes to the Automation API happen, your GitHub Action will be able to take advantage of them without a large level of
+work! And, of course, you can always write your own custom logic using the Automation API like our action does.
```

Comment:
```suggestion
The Action code is clean and concise because the power lies in the Automation API itself. The Action uses the Automation API to drive Pulumi as part of your GitHub Actions workflows. This means that as Pulumi makes
changes to the Automation API, your workflows will be able to take advantage of them as soon as they're released! And, of course, you can always write your own custom logic using the Automation API like our Action does.
```